### ISSUE TITLE
Direct download option in long press menu in queue

### ISSUE THREAD

**Issue by TippyLion28** [TeamNewPipe/NewPipe#8585]
### Checklist

- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I'm aware that this is a request for NewPipe itself and that requests for adding a new service need to be made at [NewPipeExtractor](https://github.com/TeamNewPipe/NewPipeExtractor/issues).
- [X] I have taken the time to fill in all the required details. I understand that the feature request will be dismissed otherwise.
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Feature description

Queue Long-press menu should have a download button

Basically #2171 but within the queue screen

### Why do you want this feature?

Easier than having to long press the video then go to the video info screen to download the video.

If it was easier to return to the queue from the video info screen then maybe I wouldn't care but as it stands I have to open my notification shade just to get back there

### Additional information

Sorry I have no cat pictures

**Comment by opusforlife2** [comment]
> Sorry I have no cat pictures

Here's your culprit, folks. Put 'em in jail right away. No trial needed.

**Comment by Jared234** [comment]
I can work on this.

**Comment by TippyLion28** [comment]
> I can work on this.

Godspeed

**Comment by opusforlife2** [comment]
There's no harm in assigning issues to you because they can easily be unassigned later. But make sure you're not taking on too much at a time, @Jared234.

**Comment by Jared234** [comment]
Thanks for assigning me. I am currently a student of information systems at the Technical University of Munich (TUM) and am working on NewPipe as part of an OpenSource practical course.  So, I have a lot of time to work on different issues. It will take me two weeks at most to create a PR. One of the reasons I am assigned to many issues right now is that I am still waiting for a review for many of them. 

And by the way, I really like NewPipe and its community and plan to continue working on it even after I have finished the course. 
However, if you still have any doubts in the future, please let me now so we can talk it over.

**Comment by TobiGr** [comment]
Hi and welcome, @Jared234! Thank you for taking care of this.

Just a question: are there multiple students from your course who are going to work on NewPipe or are you all working on separate projects? It might be good to know in advance if there are going to be an increased number of PRs so we can try to dedicate some workload to review your and your fellow students contributions. We got a little overwhelmed by 15+ students from the Australian National University earlier this semester and would like to be prepared better this time :)

**Comment by Jared234** [comment]
Thanks @TobiGr. In the OpenSource lab we are free to decide the projects we are working on. The only "requirement" is that we work on an OpenSource Projects with an active community. Thus we are all working on different projects. I have access to all of the issues that my classmates are working on, and can see that as of right now, nobody else has worked on NewPipe or plans to do so in the foreseeable future.

**Comment by TippyLion28** [comment]
Thank you @Jared234! Best of luck with your course :)

Have a happy and safe new year all 🎉🎉
